### PR TITLE
add custom read support filters in Delly and Manta Somatic calling

### PR DESCRIPTION
**Pull request by anoronh4**
Based on an analysis of precision and recall compared to other datasets, we are incorporating a few filters into Delly and Manta somatic structural variant outputs:
- Filter variants with any read support in the normal sample
- Filter variants that have < 5 supporting discordant reads or < 2 supporting split reads in the tumor sample

In this PR the variants will not be removed from the file entirely, instead the FILTER value will be updated with `normal_read_supp` and `tumor_read_supp`, respectively. Based on a prior analysis where we compared to pcawg calls, the estimate accuracy of the pipeline including the aforementioned filters:
- precision: 86% 
- recall: 82%
- F-score: 84
This is a large improvement over the accuracy before the filters, which is 58%, 91% and 69, respectively. 

Also the containers for strelka and manta were updated to `cmopipeline/strelka2-manta-bcftools-vt:2.0.0`. The previous container `cmopipeline/strelka2_manta:latest` is much older and does not contain bcftools or vt, even though it seems to have the same version of manta and both have python2. The container folder for `cmopipeline/strelka2_manta:latest` was removed as it is no longer necessary.

### CODE REVIEW

File: modules/process/SV/SomaticRunManta.nf [hunk 1]
```diff
@@ -47,5 +45,23 @@ process SomaticRunManta {
...
+    -s ${idTumor},${idNormal} \\
+    ${outputPrefix}.manta.vcf.gz | \\
+  bcftools filter \\
+    --soft-filter tumor_read_supp -m + \\
+    -e "FORMAT/PR[0:1] < 5 | FORMAT/SR[0:1] < 2" | \\
+  bcftools filter \\
+    --soft-filter normal_read_supp -m + \\
+    -e "FORMAT/PR[1:1] > 0 | FORMAT/SR[1:1] > 0" | \\
+  bcftools view --output-type z > \\
+    ${outputPrefix}.manta.filtered.vcf.gz
```
Comment (gongyixiao) [inline modules/process/SV/SomaticRunManta.nf:63]: Just want to make sure:

Are `${outputPrefix}.manta.vcf.gz` and `${outputPrefix}.manta.filtered.vcf.gz` having the same number of data lines? Difference are in the filtering annotations?

Same question for delly.

File: modules/process/SV/SomaticDellyCall.nf [hunk 2]
```diff
@@ -30,8 +30,21 @@ process SomaticDellyCall {
...
+    -s ${idTumor},${idNormal} \\
+    ${idTumor}__${idNormal}_${svType}.filter.bcf | \\
+  bcftools filter \\
+    --soft-filter tumor_read_supp -m + \\
+    -e "FORMAT/DV[0] < 5 | FORMAT/RV[0] < 2" | \\
+  bcftools filter \\
+    --soft-filter normal_read_supp -m + \\
+    -e "FORMAT/DV[1] > 0 | FORMAT/RV[1] > 0" | \\
+  bcftools view --output-type z > \\
+    ${idTumor}__${idNormal}_${svType}.delly.vcf.gz
```
Comment (gongyixiao) [inline modules/process/SV/SomaticDellyCall.nf:46]: If we are calling manta output as `filtered`, maybe we want to be consistant here on delly output.

File: modules/process/SV/SomaticRunManta.nf (hunk 1 above)
Comment (anoronh4) [reply modules/process/SV/SomaticRunManta.nf:63]: yes, the `--soft-filter` allows you to keep the variant, but updates the FILTER value.

File: modules/process/SV/SomaticDellyCall.nf (hunk 2 above)
Comment (anoronh4) [reply modules/process/SV/SomaticDellyCall.nf:46]: done, they are now both `*.manta.vcf.gz` and `*.delly.vcf.gz`.

**Comment by anoronh4** [comment]
__Refining Filters__
I had a few findings when comparing our calls to pcawg using the filters we have been trying out:
- No Delly calls had more than 0 alt split reads in the normal, and only a small amount of Manta calls had supporting reads in the normal, so i think this part of the filter is not necessary. However for discordant reads supporting the alt in the normal I found the following:
```
table(dat.delly$in_pcawg,dat.delly$n_delly_DV > 0 )
    FALSE TRUE
  0  4956  112
  1  7149  515
table(dat.manta$in_pcawg,dat.manta$n_manta_PR_alt > 0 )
    FALSE TRUE
  0  3682  658
  1  7343    4
```
Seems like this filter in manta is more meaningful than the equivalent filter in delly. 

For tumor support filters, seems like the split read filter is not meaningful for delly:
```
table(dat.delly$in_pcawg, dat.delly$t_delly_RV <2)  
    FALSE TRUE
  0  3478 1590
  1  5409 2255
```
And also removing a good number of true positive calls for manta:
```
table(dat.manta$in_pcawg, dat.manta$t_manta_SR_alt <2)
    FALSE TRUE
  0  3885  455
  1  6862  485
```
If we remove the normal read support filter in delly (but keep the equiv in manta), and also remove tumor split read support filter in delly, we end up with a recall of 90%, precision of 72% and F-score of 79. There is an overall loss in accuracy compared to the filters in this PR, but still a large improvement over the accuracy before this PR.

**Pull request merged**